### PR TITLE
Add target and environment to build and release

### PR DESCRIPTION
**Pull request by davidmoore1**
Move most of the build and release logic to separate build scripts
Allow user to designate targets
Implement current build and publish functions in script and allow user targets to be
implemented in user supplied gradle scripts in their own project folder

### CODE REVIEW

File: application/console/views/cron/scripts/upload/default/build.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  echo "{ \"version\" : \"${VERSION_NAME}.${VERSION_CODE}\", \"versionName\" : \"${VERSION_NAME}\", \"versionCode\" : \"${VERSION_CODE}\" } " > $OUTPUT_DIR/version.json
+  if [ -f "build_data/about/about.txt" ]; then cp build_data/about/about.txt $OUTPUT_DIR/; fi
+  PUBLISH_DIR="build_data/publish"
+  PLAY_LISTING_DIR="${PUBLISH_DIR}/play-listing"
+  LIST_DIR="${PLAY_LISTING_DIR}/"
+  MANIFEST_FILE="manifest.txt"
+  if [ -f $LIST_DIR$MANIFEST_FILE ]; then rm $LIST_DIR$MANIFEST_FILE; fi;
+  FILE_LIST=$(find $PLAY_LISTING_DIR -type f -print)
+  for f in $FILE_LIST; do fn=${f#*"$PLAY_LISTING_DIR/"}; echo $fn >> $OUTPUT_DIR/$MANIFEST_FILE; done
+  if [ -d "$PLAY_LISTING_DIR" ]; then cp -r "$PLAY_LISTING_DIR" $OUTPUT_DIR; find $OUTPUT_DIR -name whats_new.txt | while read filename; do DIR=$(dirname "${filename}"); cp "$filename" $OUTPUT_DIR; mkdir "${DIR}/changelogs"; mv "$filename" "${DIR}/changelogs/${VERSION_CODE}.txt"; done; fi
```
Comment (chrisvire) [inline application/console/views/cron/scripts/upload/default/build.sh]: Please make this multiple lines to make it easier to read.

File: application/console/views/cron/scripts/upload/default/publish.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+google_play() {
+  PAJ="${SECRETS_DIR}/playstore_api.json"
+  cd $ARTIFACTS_DIR
+  set +x
+  if [ -z "$PROMOTE_FROM" ]; then	fastlane supply -j $PAJ -b *.apk -p $(cat package_name.txt) --track $CHANNEL -m play-listing; else fastlane supply -j $PAJ -b *.apk -p $(cat package_name.txt) --track $PROMOTE_FROM --track_promote_to $CHANNEL -m play-listing; fi
```
Comment (chrisvire) [inline application/console/views/cron/scripts/upload/default/publish.sh]: Please make this multiple lines for easier reading.

File: application/console/views/cron/scripts/upload/default/build.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  if [ -f $LIST_DIR$MANIFEST_FILE ];
+    then rm $LIST_DIR$MANIFEST_FILE;
+  fi
+  FILE_LIST=$(find $PLAY_LISTING_DIR -type f -print)
+  for f in $FILE_LIST; do fn=${f#*"$PLAY_LISTING_DIR/"};
+    echo $fn >> $OUTPUT_DIR/$MANIFEST_FILE;
+  done
+  if [ -d "$PLAY_LISTING_DIR" ]; then
+    cp -r "$PLAY_LISTING_DIR" $OUTPUT_DIR;
+    find $OUTPUT_DIR -name whats_new.txt | while read filename; do DIR=$(dirname "${filename}");
```
Comment (chrisvire) [inline application/console/views/cron/scripts/upload/default/build.sh]: ` DIR=$(dirname "${filename}");` should go on next line.  Don't need semi-colons anymore

File: application/console/views/cron/scripts/upload/default/build.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  LIST_DIR="${PLAY_LISTING_DIR}/"
+  MANIFEST_FILE="manifest.txt"
+  if [ -f $LIST_DIR$MANIFEST_FILE ]; then
+    rm $LIST_DIR$MANIFEST_FILE
+  fi
+  FILE_LIST=$(find $PLAY_LISTING_DIR -type f -print)
+  for f in $FILE_LIST
+  do 
+    fn=${f#*"$PLAY_LISTING_DIR/"}
+    echo "$fn" >> "$OUTPUT_DIR"/$MANIFEST_FILE
```
Comment (chrisvire) [inline application/console/views/cron/scripts/upload/default/build.sh]: @davidmoore1: I wasn't so sure about this one.  I would check the output of the MANIFEST_FILE

File: application/console/views/cron/scripts/upload/default/build.sh [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  done
+  if [ -d "$PLAY_LISTING_DIR" ]; then
+    cp -r "$PLAY_LISTING_DIR" "$OUTPUT_DIR"
+    find "$OUTPUT_DIR" -name whats_new.txt | while read -r filename
```
Comment (chrisvire) [inline application/console/views/cron/scripts/upload/default/build.sh]: @davidmoore1: shellcheck suggested adding `-r` option

File: application/console/views/cron/scripts/upload/default/build.sh (hunk 4 above)
Comment (davidmoore1) [reply application/console/views/cron/scripts/upload/default/build.sh]: Looked fine I think {"files":["en-US/images/phoneScreenshots/Screen","Shot","2018-04-17","at","4.02.23","PM.png","en-US/images/phoneScreenshots/Screen","Shot","2018-04-19","at","2.29.42","PM.png","en-US/images/icon.png","en-US/images/featureGraphic.png","en-US/full_description.txt","en-US/short_description.txt","en-US/title.txt"],"default-language":"en-US"}

**Pull request merged**